note_taking-explainer.md

```diff
@@ -0,0 +1,129 @@
+
+By using a dictionary we can easily add other note-taking app functionality or
+extend the current functionality, and it keeps the manifest organised with
+related functionality together. It also means note-taking can be specified
+largely orthogonally to other manifest specification changes, which helps to
+keep the manifest specification modular and manageable.
+
+## Alternatives Considered
+
+A common theme of alternatives is to make a generalised interface for web apps
+to advertise their capabilities. Ideally this would let us solve a whole class
+of problems instead of just one case. For example, a similar mechanism to "
+create a new note" could be used to "create new calendar entry" or "add a
+contact".
+
+### Extend the `Shortcuts` Member
+
+If we used the existing [Shortcuts manifest member](
+https://www.w3.org/TR/appmanifest/#shortcuts-member), we would need to add a way
+for the user agent to identify shortcuts as intended to be used for specific
+purposes/tasks (eg. a new optional "purpose" field on each shortcut entry, with
+a defined set of values like "new-note", "new-contact", etc). Some of these
+future tasks might need additional parameters/context, which would further
+extend and complicate the `Shortcuts` member. Additionally, Shortcuts are
+intended to be displayed to the user, while many tasks/capabilities should not
+be displayed in a shortcuts menu, eg. any parameterised or contextual tasks,
+non-task capabilities like [lock screen](https://github.com/WICG/lock-screen)
+support. It seems like extending `Shortcuts` this far would be significantly
+changing the semantics of the member.
+
+### A General Capability Member
+
+There are existing general capabilities approaches for other functionality of
+web apps/pages:
+
+* [Web Share Target](https://w3c.github.io/web-share-target/)
+* [File Handling](https://github.com/WICG/file-handling/blob/main/explainer.md)
+* [Protocol Handlers](https://html.spec.whatwg.org/multipage/system-state.html#dom-navigator-registerprotocolhandler)
+* [Declarative Web Actions](https://github.com/slightlyoff/declarative_web_actions)
+
+So perhaps we could do something similar with a more general-purpose field like:
+
+```
+"integrations":  [
+  { "task": "new-note", "url": <URL> }
+  { "task": "add-to-existing-note", "url": <URL>, <additional params> }
+  { "task": "lock-screen", "url": <URL> }
+  { "task": "new-contact", "url": <URL>, <probably additional params> }
+]
+```
+
+Or:
+
+```
+"integrations": {
+  "note-taking": {
+    "create": {
+      "action": "/new_note.html",
+    },
+    "add-item": {
+      "action": "/add_item_to_note.html",
+      "params": {
+        "title": "name",
+        "text":  "description",
+        "url":   "link"
+      }
+    }
+  }
+}
+```
+
+Where each separate integration has a particular schema of fields.
+
+However, these individual integration likely need to be specified individually
```

Comment:
There's a bit of a missing link here: this document is roughly saying there are two alternatives:

- The basic `new_note_url` being proposed.
- The "over-engineered" integrations -> note-taking -> create mentioned here.

However, the perhaps obvious but instated fact is that the former can be extended with all of these features. For example, `note-taking` has `new-note-url` but that could also get an `add-item` section later, just like what you've got here. The fact that you _only_ mention `add-item` in the "alternatives considered" section implies that we won't be able to do it with your proposal.

It might be worth adding a "future considerations" section, saying that we could add `add-item` and maybe also a `new-note-with-contents` that would let a voice assistant take a new note and populate it with some spoken text.

Reply:
I tried to get that across when talking about the proposal but I agree it's not stated very obviously, so added a section as you suggested.